ClipboardAPI/clipboard-change-event-explainer.md

```diff
@@ -0,0 +1,190 @@
+```
+
+## 5. Event spec details and open questions
+
+### 5.1 User permission requirement
+
+#### 5.1.1 Approach 1 - clipboard-read permission required to listen to clipboardchange event
+Since the clipboard contains privacy-sensitive data, we should protect access to the clipboard change event using a user permission - clipboard-read. The web author should ensure that the site has the permission before it starts listening to this event otherwise the provided event handler won't be invoked whenever the clipboard changes. To check if the current user has clipboard-read permissions for the site, the [query](https://www.w3.org/TR/permissions/#query-method) method of the [Permissions API](https://www.w3.org/TR/permissions/#permissions-api) can be used. We should consider logging a warning message if the web author starts listening to clipboardchange without acquiring the permissions since web developers might miss integrating the permissions flow into their user experience.
+
+Web apps can request for the clipboard-read permissions by performing a read operation using one of [read](https://w3c.github.io/clipboard-apis/#dom-clipboard-read) or [readText](https://w3c.github.io/clipboard-apis/#dom-clipboard-readtext) methods of the [Async clipboard API](https://w3c.github.io/clipboard-apis/#async-clipboard-api). 
+
+In future, we may have additional ways of prompting the user for permissions - 1) By explicitly requesting for "clipboard-read" permission, the API for this is still under discussion (https://github.com/w3c/permissions/issues/158). 2) The user can be prompted for permissions as soon as the "addEventListener" method is called with "clipboardchange" in case the permissions are not already granted. This is still open for discussion as it is not a common pattern to prompt user for permissions when attaching event listeners.
+
+##### Pros
+1. This approach aligns with the security model for other clipboard APIs and similar APIs that allow access to sensitive information.
+2. Ensures user consent before accessing clipboard data.
+
+##### Cons
+1. This approach imposes a small restriction on web authors as they have to call a clipboard read API method before starting to listen for the clipboardchange event.
+
+#### 5.1.2 Approach 2 - No permission required
+Since no data is being sent as part of the clipboardchange event, it can be argued that we don't need any permission to simply know when clipboard contents change. This will simplify the user flow as they don't need to explicitly ask for permissions before listening to the event.
+
+##### Pros
+1. Simpler implementation and user experience
+
+##### Cons
+1. Open to privacy attacks which simply monitor clipboard changes without user consent.
+Example - Untrusted sites can perform phishing attacks like - A user copies a URL to share with a friend, but the clipboard content is replaced with a phishing link that looks similar to the original URL.
```

Comment:
I think the main con is simply that a site will be able to monitor the clipboard - a privileged OS component - without the user explicitly allowing it. That may be interpreted as a user privacy violation. It's also worth noting that the author would only know that the clipboard changed, but wouldn't have any details about the payload.